lib/utilities/MD5Sum.js

```diff
@@ -0,0 +1,24 @@
+const Transform = require('stream').Transform;
+const crypto = require('crypto');
+
+class MD5Sum extends Transform {
```

Comment:
Can you add a comment here, about the design of the done callback, saying the done callback need to be synchronous and will be called before the stream end to avoid any strange behavior with this :)